mlir/include/mlir/Dialect/Linalg/IR/LinalgStructuredOps.td

```diff
@@ -587,7 +590,8 @@ def ElementwiseOp : LinalgStructuredBase_Op<"elementwise", [
       Variadic<AnyType>:$inputs,
       Variadic<AnyShaped>:$outputs,
       ElementwiseKindAttr:$kind,
-      DefaultValuedOptionalAttr<AffineMapArrayAttr, "{}">:$indexing_maps
+      DefaultValuedOptionalAttr<AffineMapArrayAttr, "{}">:$indexing_maps,
+      DefaultValuedOptionalAttr<TypeFnAttr, "TypeFn::cast_signed">:$cast
```

Comment:
I think it would be better to make this a list of `TypeFnAttr` which allows for a sentinel `none` if no castin is required. Some parser/printer helpers can allow something like `[cast_signed, -]` to say no casting is needed. I'd also say this list should be off the size of the number of `ins` operands.